Sources/SwiftDocC/Semantics/Metadata/SourceLanguage+Directive.swift

```diff
@@ -0,0 +1,27 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+// A custom function for parsing `SourceLanguage` as a directive argument.
+//
+// Initializer chaining with the `??` coalescing operator is not a supported Swift feature,
+// so instead of extending ``SourceLanguage`` to conform to ``DirectiveArgumentValueConvertible``,
+// define a custom function to parse the argument value.
+func parseSourceLanguage(_ bundle: DocumentationBundle, _ argumentValue: String) -> SourceLanguage? {
```

Comment:
The `bundle` parameter is unused and the `argumentValue` parameter is not very descriptive (and could benefit) from a label.

Reply:
Actually, this checking `SourceLanguage(knownLanguageIdentifier:)` and then falling back to `SourceLanguage(id:)` is pointless because `SourceLanguage(id:)` already returns the known language if the argument matches known id alias.